segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -641,7 +643,30 @@ public void truncateSegment(TruncateSegment truncateSegment) {
         log.info(truncateSegment.getRequestId(), "Truncating segment {} at offset {}.",
                 segment, offset);
         segmentStore.truncateStreamSegment(segment, offset, TIMEOUT)
-                .thenAccept(v -> connection.send(new SegmentTruncated(truncateSegment.getRequestId(), segment)))
+                .whenComplete((properties, exp) -> {
+                    if (exp != null) {
+                        handleException(truncateSegment.getRequestId(), segment, offset, operation, exp);
+                    } else {
+                        segmentStore.getStreamSegmentInfo(getIndexSegmentName(segment), PravegaRequestProcessor.TIMEOUT)
+                                .whenComplete((p, ex) -> {
+                                    if (ex != null) {
+                                        handleException(truncateSegment.getRequestId(), getIndexSegmentName(segment), offset, "truncateIndexSegment", ex);
+                                    } else {
+                                        try {
+                                            long endIndex = p.getLength();
+                                            //TO-DO
```

Comment:
please add a proper TODO: statement that needs to be clarified, so that it can be verified.

Reply:
Done